test/index.test.js

```diff
@@ -1,55 +1,640 @@
-'use strict';
+'use strict'
 
-const { test } = require('tap');
-const nock = require('nock');
-const { run } = require('../src/run');
+const { test } = require('tap')
+const nock = require('nock')
+const { run } = require('../src/run')
 
-const TOKEN = 'jifeo2903u089jf3920';
-const TEAM_ID = 'i992-j9f23j09-j092j0';
+// Import fixtures
+const pushPayload = require('./fixtures/push')
+const pullRequestPayload = require('./fixtures/pull_request')
+const deploymentPayload = require('./fixtures/deployment')
+const deploymentStatusPayload = require('./fixtures/deployment_status')
+const statusPayload = require('./fixtures/status')
+const packagePayload = require('./fixtures/package')
+const releasePayload = require('./fixtures/release')
+const issuePayload = require('./fixtures/issue')
+
+// These params are used for integration testing.
+const { NOCK_ENABLED, OPS_PLATFORM_TEST_TOKEN } = process.env
+
+const TOKEN = OPS_PLATFORM_TEST_TOKEN || 'jifeo2903u089jf3920'
+const TEAM_ID = 'i992-j9f23j09-j092j0'
 
 const setInput = (name, value) => {
-  process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`]=value;
+  process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`] = value
 }
 
 const clearInput = (name) => {
-  delete process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`];
+  delete process.env[`INPUT_${name.replace(/ /g, '_').toUpperCase()}`]
 }
 
 test('Push event with only required fields has additional data added', async ({ is }) => {
-  setInput('token', TOKEN);
-  setInput('team_id', TEAM_ID);
-  setInput('event_name', 'deployment');
-  setInput('event_action', 'succeeded');
+  setInput('token', TOKEN)
+  setInput('team_id', TEAM_ID)
+  setInput('event_name', 'deployment')
+  setInput('event_action', 'succeeded')
 
   const context = {
     eventName: 'push',
-    payload: {
-      ref: 'refs/heads/main',
-      after: '12345',
-      repository: {
-        full_name: 'org/test'
-      }
-    }
-  };
-
-  const req = nock('https://events.cto.ai')
-    .post('/', {
-      team_id: TEAM_ID,
-      event_name: 'deployment',
-      event_action: 'succeeded',
-      environment: null,
-      image: null,
-      branch: 'main',
-      commit: '12345',
-      repo: 'org/test'
-    })
-    .reply(200, { message: 'event written', data: {} });
-
-  await run(context);
-  is(req.isDone(), true);
-
-  clearInput('token');
-  clearInput('team_id');
-  clearInput('event_name');
-  clearInput('event_action');
+    payload: pushPayload
+  }
+
+  if (NOCK_ENABLED) {
+    nock('https://events.cto.ai')
+      .matchHeader('authorization', `Bearer ${TOKEN}`)
+      .post('/', {
+        team_id: TEAM_ID,
+        event_name: 'deployment',
+        event_action: 'succeeded',
+        environment: null,
+        image: null,
+        branch: 'simple-tag',
+        commit: '6113728f27ae82c7b1a177c8d03f9e96e0adf246',
+        repo: 'Codertocat/Hello-World'
+      })
+      .reply(200, { message: 'event written', data: {} })
```

Comment:
sorry one question: does this mock response data mock what an actual API call would respond with?

Reply:
It mocks what is returned from `events` which for this type of payload is that `message` and `data` which includes the write receipt from the data base. I didn't mock that side to avoid unnecessary complexity and database details to the public.